lib/pure/math.nim

```diff
@@ -330,6 +226,57 @@ else:
     var y = exp(2.0*x)
     return (y-1.0)/(y+1.0)
 
+proc round*(x: float, places: int = 0): float =
+  ## Round a floating point number.
+  ##
+  ## If `places` is 0 (or omitted), round to the nearest integral value
+  ## following normal mathematical rounding rules (e.g. `round(54.5) -> 55.0`).
+  ## If `places` is greater than 0, round to the given number of decimal
+  ## places, e.g. `round(54.346, 2) -> 54.35`.
+  ## If `places` is negative, round to the left of the decimal place, e.g.
+  ## `round(537.345, -1) -> 540.0`
+  if places == 0:
+    result = round0(x)
+  else:
+    var mult = pow(10.0, places.float)
+    result = round0(x*mult)/mult
+
+when not defined(JS):
+  proc frexp*(x: float, exponent: var int): float {.
+    importc: "frexp", header: "<math.h>".}
+    ## Split a number into mantissa and exponent.
+    ## `frexp` calculates the mantissa m (a float with 0.5 <= abs(m) < 1) and
+    ## the integer value n such that `x` (the original float value) equals
+    ## m * 2**n. frexp stores n in `exponent` and returns m.
+else:
+  proc frexp*(x: float, exponent: var int): float =
+    if x == 0.0:
+      exponent = 0
+      result = 0.0
+    elif x < 0.0:
+      result = -frexp(-x, exponent)
+    else:
+      var ex = floor(log2(x))
+      exponent = round(ex)
+      result = x / pow(2.0, ex)
+
+proc split*(x: BiggestFloat): tuple[intpart: BiggestFloat, floatpart: BiggestFloat] =
```

Comment:
Don't name this 'split' please. Name it 'intfract' or something better than that.